### ISSUE TITLE
SDL_Vulkan_LoadLibrary failing on Linux

### ISSUE THREAD

**Issue by WayneSherman** [BeRo1985/pasvulkan#12]
Tested on Kubuntu 18.04 64-bit w/ [updated video stack](https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers)

In PasVulkan.Application.pas line 7828, the call to SDL_Vulkan_LoadLibrary is failing with "Video subsystem has not been initialized".  Also, the return value is  not checked:
```
if fSDLVersionWithVulkanSupport then begin
  SDL_Vulkan_LoadLibrary(VK_DEFAULT_LIB_NAME);
end;
```
If it is important for this function to succeed, the return value should be checked:
```
if fSDLVersionWithVulkanSupport then begin
  if SDL_Vulkan_LoadLibrary(VK_DEFAULT_LIB_NAME)<0 then begin
    raise EpvApplication.Create('SDL','SDL_Vulkan_LoadLibrary failed: '+SDL_GetError,LOG_ERROR);
  end;
 end;
```

**Comment by BeRo1985** [comment]
No, it is not important that it succeeds, because in this case PasVulkan should use the fallback code path then, where PasVulkan will handle all Vulkan-specific code itself. See the fix at https://github.com/BeRo1985/pasvulkan/commit/9f50ff49cfaae651a7fca9e2e6a9d804b73b1527

**Comment by BeRo1985** [comment]
Because not all newer SDL2 libary binaries in some Linux distros are compiled with Vulkan support builtin, because it is a make-parameter-optional feature of SDL2.

**Comment by WayneSherman** [comment]
SDL_Vulkan_LoadLibrary(VK_DEFAULT_LIB_NAME) cannot be called before the video subsystem is initialized.  So it should be called after SDL_Init.  It then succeeds:
[Reference](https://wiki.libsdl.org/SDL_VideoInit)

diff --git a/src/PasVulkan.Application.pas b/src/PasVulkan.Application.pas
index cb364c1..0854ed6 100644
--- a/src/PasVulkan.Application.pas
+++ b/src/PasVulkan.Application.pas
@@ -7814,23 +7814,6 @@ begin
  ReadConfig;
 
 {$if defined(PasVulkanUseSDL2)}
- SDL_GetVersion(fSDLVersion);
-
-{$if defined(PasVulkanUseSDL2WithVulkanSupport)}
- fSDLVersionWithVulkanSupport:=(fSDLVersion.Major>=3) or
-                               ((fSDLVersion.Major=2) and
-                                (((fSDLVersion.Minor=0) and (fSDLVersion.Patch>=6)) or
-                                 (fSDLVersion.Minor>=1)
-                                )
-                               );
-
- if fSDLVersionWithVulkanSupport then begin
-  SDL_Vulkan_LoadLibrary(VK_DEFAULT_LIB_NAME);
- end;
-{$ifend}
-{$ifend}
-
-{$if defined(PasVulkanUseSDL2)}
  if fAndroidSeparateMouseAndTouch then begin
   SDL_SetHint(SDL_HINT_ANDROID_SEPARATE_MOUSE_AND_TOUCH,'1');
  end else begin
@@ -7858,6 +7841,25 @@ begin
 {$else}
 {$ifend}
 
+{$if defined(PasVulkanUseSDL2)}
+ SDL_GetVersion(fSDLVersion);
+
+{$if defined(PasVulkanUseSDL2WithVulkanSupport)}
+ fSDLVersionWithVulkanSupport:=(fSDLVersion.Major>=3) or
+                               ((fSDLVersion.Major=2) and
+                                (((fSDLVersion.Minor=0) and (fSDLVersion.Patch>=6)) or
+                                 (fSDLVersion.Minor>=1)
+                                )
+                               );
+
+ if fSDLVersionWithVulkanSupport then begin
+  if SDL_Vulkan_LoadLibrary(VK_DEFAULT_LIB_NAME)<0 then begin
+    raise EpvApplication.Create('SDL','SDL_Vulkan_LoadLibrary failed: '+SDL_GetError,LOG_ERROR);
+  end;
+ end;
+{$ifend}
+{$ifend}
+
 {$if defined(Unix) and not defined(Android)}
  InstallSignalHandlers;
 {$ifend}

**Comment by WayneSherman** [comment]
>Because not all newer SDL2 libary binaries in some Linux distros are compiled with Vulkan support >builtin, because it is a make-parameter-optional feature of SDL2.

Yes, that is what I discovered with Ubuntu 18.04.  It is fixed for 18.10 and in progress for 18.04:  https://bugs.launchpad.net/ubuntu/+source/libsdl2/+bug/1740517

Until it gets fixed for Ubuntu 18.04, this PPA has SDL2 with Vulkan enabled:
https://launchpad.net/~andreasschultes/+archive/ubuntu/libsdl2-vulkan

**Comment by WayneSherman** [comment]
Yes, this new commit is working for me.  Thanks.
